src/ranged.cpp

```diff
@@ -192,12 +192,9 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
         trajectory = g->m.find_clear_path( source, target );
     }
 
+    static emit_id muzzle_smoke( "emit_smoke_plume" );
     if( proj_effects.count( "MUZZLE_SMOKE" ) ) {
-        for( const auto& e : closest_tripoints_first( 1, trajectory.empty() ? source : trajectory[ 0 ] ) ) {
-            if( one_in( 2 ) ) {
-                g->m.add_field( e, fd_smoke, 1, 0 );
-            }
-        }
+        g->m.emit_field( trajectory.empty() ? source : trajectory[ 0 ], muzzle_smoke );
```

Comment:
The ternary won't be required if you move this line below [that one](https://github.com/CleverRaven/Cataclysm-DDA/pull/17530/files#diff-c5873ed460ae46963dac594829494e52R210). Also, `trajectory.front()` is cleaner.